### DISCUSSION TITLE
Why can't we use a boolean shared value with enabled property?

### DISCUSSION THREAD

**Discussion by Glazzes** [software-mansion/react-native-gesture-handler#2823] [Q&A]
I've been trying to disable gestures in the middle of a snapback animation using React state values, however in low end android devices updating the state on the middle of this particular animation slightly freezes it, not much but it's noticeable to me, I wonder why can't we use a shared value to disable gestures? Like this:

```js
const gestureEnabled = useSharedValue<boolean>(true);

const pan = Gesture.Pan()
  .enabled(gestureEnabled)
```

**Comment by JCYDigitalinnov** [comment]
Same question, that should be possible.

**Comment by doublelam** [comment]
Same question, setState to change the enabled value and pass a new instance is not quick enough and it makes animation flickering and slow.
Or give a method to change the value but not a new instance like: 

```
const pan = Gesture.Pan();
```

somewhere else:
```
pan.enabled(false);  // do not return a new instance so that I don't need to pass it to the component once more
```
